src/Marten/LinqParsing.cs

```diff
@@ -136,4 +140,25 @@ private IMethodCallParser determineMethodParser(MethodCallExpression expression)
         return MethodCallParsers.FirstOrDefault(x => x.Matches(expression))
                ?? _parsers.FirstOrDefault(x => x.Matches(expression));
     }
+
+    /// <summary>
+    ///     https://learn.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.metadatatoken?view=net-8.0
+    ///     MetadataToken -- "A value which, in combination with Module, uniquely identifies a metadata element."
+    /// </summary>
+    private static string ToKey(MethodInfo expressionMethod)
+    {
+        Encoutered.Add(
+            $"{expressionMethod.Module.Name}_{expressionMethod.Module.MetadataToken}_{expressionMethod.MetadataToken}_{expressionMethod.DeclaringType?.Name}_{expressionMethod.Name}");
+
+        Console.WriteLine();
+        Console.WriteLine();
+        Console.WriteLine("---");
+        Console.WriteLine();
+        foreach (var x in Encoutered.ToArray())
+        {
+            Console.WriteLine(x);
+        }
+
+        return $"{expressionMethod.Module.Name}_{expressionMethod.MetadataToken}";
```

Comment:
The fix -- MetadataToken seems to be unique per expressionMethod.Module.Name in my tests..
(expressionMethod.Module.MetadataToken isnt unique for me..?)